generators/app/templates/src/projectName.jsx

```diff
@@ -16,33 +26,29 @@ class <%= projectClassName %> extends React.Component {
   }
 
   render() {
+    const { name, variant, ...customProps } = this.props;
     const classes = classNames(['<%= projectCssClassName %>',
-      this.props.variant,
+      variant,
       this.state.isSelected && 'u-selected',
+      customProps.className,
     ]);
 
 
-    if (!this.props.name) {
+    if (!name) {
       return null;
     }
-    if (!this.props.variant) {
+    if (!variant) {
       return null;
     }
     return (<button
+      {...customProps}
```

Comment:
Yes it would be overwritten in the next line, but you'll notice, up on line 33 i'm adding the passed in classNames to the classes object, so they are still getting applied. I pulled them out since we're modifying them. We could just as well update classNames on the customProps object, but I thought this was a touch more explicit.